thewalrus/quantum.py

```diff
@@ -1,1300 +1,1320 @@
+        rtol (float): the relative tolerance parameter used in `np.allclose`
+        atol (float): the absolute tolerance parameter used in `np.allclose`
+
+    Return:
+        (array): Tensor containing the Fock representation of the Gaussian unitary
+    """
+    # Check the matrix is symplectic
+    if check_symplectic:
+        if not is_symplectic(S, rtol=rtol, atol=atol):
+            raise ValueError("The matrix S is not symplectic")
+
+    # And that S and alpha have compatible dimensions
+    m, _ = S.shape
+    l = m // 2
+    if l != len(alpha):
+        raise ValueError(
+            "The matrix S and the vector alpha do not have compatible dimensions"
+        )
+    # Check if S corresponds to an interferometer, if so use optimized routines
+    if np.allclose(S @ S.T, np.identity(m), rtol=rtol, atol=atol) and np.allclose(
+        alpha, 0, rtol=rtol, atol=atol
+    ):
+        reU = S[:l, :l]
+        imU = S[:l, l:]
+        if np.allclose(imU, 0, rtol=rtol, atol=atol):
+            Ub = np.block([[0 * reU, -reU], [-reU.T, 0 * reU]])
+            tensor = interferometer_real(Ub, cutoff)
+        else:
+            U = reU - 1j * imU
+            Ub = np.block([[0 * U, -U], [-U.T, 0 * U]])
+            tensor = interferometer(Ub, cutoff)
+    else:
+        # Construct the covariance matrix of l two-mode squeezed vacua pairing modes i and i+l
+        ch = np.cosh(choi_r) * np.identity(l)
+        sh = np.sinh(choi_r) * np.identity(l)
+        zh = np.zeros([l, l])
+        Schoi = np.block(
+            [[ch, sh, zh, zh], [sh, ch, zh, zh], [zh, zh, ch, -sh], [zh, zh, -sh, ch]]
+        )
+        # And then its Choi expanded symplectic
+        S_exp = expand(S, list(range(l)), 2 * l) @ Schoi
+        # And this is the corresponding covariance matrix
+        cov = S_exp @ S_exp.T
+        alphat = np.array(list(alpha) + ([0] * l))
+        x = 2 * alphat.real
+        p = 2 * alphat.imag
+        mu = np.concatenate([x, p])
+
+        tensor = state_vector(
+            mu,
+            cov,
+            normalize=False,
+            cutoff=cutoff,
+            hbar=2,
+            check_purity=False,
+            choi_r=choi_r,
+        )
+
+    if sf_order:
+        sf_indexing = tuple(chain.from_iterable([[i, i + l] for i in range(l)]))
+        return tensor.transpose(sf_indexing)
+
+    return tensor
+
+
+def probabilities(mu, cov, cutoff, parallel=False, hbar=2.0, rtol=1e-05, atol=1e-08):
+    r"""Generate the Fock space probabilities of a Gaussian state up to a Fock space cutoff.
+
+    .. note::
+
+        Parallization is already being done by OpenMP when calling ``density_matrix_element``.
+        To get a speed-up from using ``parallel=True`` it must be turned off by setting the
+        environment variable ``OMP_NUM_THREADS=1`` (forcing single threaded use). Remove the
+        environment variable or set it to ``OMP_NUM_THREADS=''`` to again use multithreading
+        with OpenMP.
```

Comment:
Attempted to make it slightly clearer, let me know what you think!

```suggestion
        Individual density matrix elements are computed using multithreading by OpenMP.
        Setting ``parallel=True`` will further result in *multiple* density matrix elements
        being computed in parallel.

        When setting ``parallel=True``, OpenMP will be turned off by setting the
        environment variable ``OMP_NUM_THREADS=1`` (forcing single threaded use for individual
        matrix elements). Remove the environment variable or set it to ``OMP_NUM_THREADS=''``
        to again use multithreading with OpenMP.
```

Reply:
Thanks Josh. It looks good. :+1: Just made some small changes to the last line.